newIDE/app/src/Credits/CreditsStatusBanner.js

```diff
@@ -0,0 +1,89 @@
+// @flow
+import { Trans } from '@lingui/macro';
+import * as React from 'react';
+import { Column } from '../UI/Grid';
+import GDevelopThemeContext from '../UI/Theme/GDevelopThemeContext';
+import { LineStackLayout, ResponsiveLineStackLayout } from '../UI/Layout';
+import FlatButton from '../UI/FlatButton';
+import Text from '../UI/Text';
+import AuthenticatedUserContext from '../Profile/AuthenticatedUserContext';
+import CreditsPackagesDialog from './CreditsPackagesDialog';
+import Coin from './Icons/Coin';
+
+const styles = {
+  container: {
+    borderRadius: 8,
+    padding: 8,
+  },
+};
+
+type Props = {|
+  displayPurchaseAction: boolean,
+|};
+
+const CreditsStatusBanner = ({ displayPurchaseAction }: Props) => {
+  const gdevelopTheme = React.useContext(GDevelopThemeContext);
+  const authenticatedUser = React.useContext(AuthenticatedUserContext);
+  const [
+    creditsPackDialogOpen,
+    setCreditsPackDialogOpen,
+  ] = React.useState<boolean>(false);
+
+  // Ensure credits are refreshed when this component is shown.
+  React.useEffect(
+    () => {
+      authenticatedUser.onRefreshLimits();
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I think I tried once to do:
```js
const { onRefreshLimits } = authenticatedUser;
React.useEffect(..., [onRefreshLimits])
```
And it worked well because onRefreshLimits does not change when the authenticated user object does

Reply:
indeed!